src/passes/RemoveUnusedBrs.cpp

```diff
@@ -513,6 +513,10 @@ struct RemoveUnusedBrs : public WalkerPass<PostWalker<RemoveUnusedBrs>> {
             auto* rep = getDroppedChildrenAndAppend(
               curr, wasm, getPassOptions(), br, DropMode::IgnoreParentEffects);
             replaceCurrent(rep);
+            // We modified the code here and may have added a drop, etc. Leave
+            // seeing if things can flow out of the new code to later
+            // iterations.
+            stopFlow();
```

Comment:
I'm not very familiar with the algorithm here. Why would a `drop` in particular interfere with the flow? (And what exactly is "flowing out"? Does that mean we branch to the innermost block so the `br` is not necessary?)

Also maybe because my English is bad but it's hard to parse the comments... What does "Leave seeing if .." mean?

Reply:
The idea is that a br can be removed if it is in the right place:
```wat
(block $x (result i32)
  (br_if $x (..value..))
)
```
We can turn that into
```wat
(block $x (result i32)
  (..value..)
)
```
We just flow out of the block with or without the br here. But if there is any code in the middle, we can't, like this:
```wat
(block $x (result i32)
  (br_if $x (..value..))
  (return)
)
```
"Leave seeing if" here was meant to say that we leave any further work for later iterations. I rewrote it now.

Reply:
So is this possible only when `br`'s depth is 0, so any structure is created in between is blocking the flow now? Then is `drop` somehow special in that regard or any structure would block it the same?

Reply:
Yes, depth 0 is necessary I think. Drop isn't special, also a `nop` would block it. Most things do. But some code doesn't, e.g.
```
(block $b
  (if
    (..condition..)
    (br $b) ;; this can be removed
    (return)
  )
)
```
The `if` is between the `br` and the `block`, but it is ok. This stuff is handled in `visitAny` where we call `stopFlow` in the proper places (see e.g. the `if` handling there).